src/Cart/Mixin.js

```diff
@@ -21,6 +21,8 @@ export default {
         cartDiscountPercentage() {
             if (!this.cartCollection.discount.percentage) return 0.00
 
+            if (this.cartCollection.discount.percentage === '100.00') this.currentCheckout.payment = { provider: 'free' }
```

Comment:
Need to add a fallback here to reset this flag if the discount is removed subsequently.